src/async_op.c

```diff
@@ -35,30 +41,69 @@ static void async_op_dispose(ASYNC_OP* async_op)
 #define IS_POWER_OF_2(x) \
     (((x) & ((x) - 1)) == 0)
 
+/*tiny function that returns the next multiple of "align" starting from "start" (included). Does 1 mod and 1 branch*/
+static void* get_next_aligned_address(void* start, uint32_t align)
+{
+    void* result;
+    uint32_t m = (uintptr_t)start % align;
+    if (m == 0)
+    {
+        result = start;
+    }
+    else
+    {
+        result = (void*)((uintptr_t)start - m + align);
+    }

```

Comment:
Consider replacing with `result = (void*)((uintptr_t)start - m + m ? align : 0)`

Reply:
Nice comment! Made me think for a moment, and I decided to go with a multiplication. The compiler can go bonkers with optimizations as much as it wants.